pkg/reconciler/common/deployments.go

```diff
@@ -52,7 +55,7 @@ func CheckDeployments(ctx context.Context, manifest *mf.Manifest, instance base.
 
 	if len(nonReadyDeployments) > 0 {
 		status.MarkDeploymentsNotReady(nonReadyDeployments)
-		return nil
+		return controller.NewRequeueAfter(1 * time.Second)
```

Comment:
Do we need to requeue blindly ? The controllers are setting up informers for deployments, so the resource will get requeued anyway when they change

https://github.com/knative/operator/blob/59db3c03435265f475b5eea80595f81d4aa8a0a7/pkg/reconciler/knativeeventing/controller.go#L71-L74

https://github.com/knative/operator/blob/59db3c03435265f475b5eea80595f81d4aa8a0a7/pkg/reconciler/knativeserving/controller.go#L71-L74

Reply:
and if that's not happening, then the informer handling is where we need to tweak to fix the issue

Reply:
I am aware of that, I wanted to make sure it happens faster in case something is slow.

Reply:
The other thing is that if I dont re-enqueue there and block the logic we still face the proble,

Reply:
These are the stages:

```
	stages := common.Stages{
		common.AppendTarget,
		ingress.AppendTargetIngress,
		security.AppendTargetSecurity,
		common.AppendAdditionalManifests,
		r.appendExtensionManifests,
		r.transform,
		manifests.Install,
		common.CheckDeployments,  // <- Make sure we are up here otherwise don't progress bellow
		common.InstallWebhookConfigs,
		common.InstallWebhookDependentResources,
		manifests.SetManifestPaths,
		common.MarkStatusSuccess,
		common.DeleteObsoleteResources(ctx, ks, r.installed),
	}
```

Reply:
I could add a check with a back off strategy at the controller side where the informer gets notified by deployments as well.